plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt

```diff
@@ -173,12 +175,53 @@ internal class LocationPuckManager(
     vararg bearings: Double,
     options: (ValueAnimator.() -> Unit)? = null,
     forceUpdate: Boolean = false
+  ) {
+    when (settings2.puckBearingEnabled) {
+      false -> {
+        // Puck bearing is disabled, snap back to north.
+        snapToNorth(options, false)
+      }
+      true -> {
+        // Puck bearing is enabled, animate to new bearing.
+        animateToBearing(bearings, options, forceUpdate)
+      }
+    }
+  }
+
+  @VisibleForTesting(otherwise = PRIVATE)
+  fun snapToNorth(
```

Comment:
just as a thought - why can't we actually re-use `animateToBearing` function with options.duration = 0 and target = 0?

Reply:
hmmm I guess we could do so